### ISSUE TITLE
[Request] MCH: Replace Hypercharge with Barrel Stabilizer under conditions.

### ISSUE THREAD

**Issue by fineanddandy26** [MKhayle/XIVComboExpanded#423]
I think it'd be useful as an Accessibility or perhaps Secret combo to replace Hypercharge with Barrel Stabilizer when Heat Gauge is less than 50 and Barrel Stabilizer is off cooldown.

**Comment by fineanddandy26** [comment]
And perhaps an alternate combo in Accessibility/Secret that makes it so Barrel Stabilizer ALWAYS replaces Hypercharge when it's off cooldown.

**Comment by kaedys** [comment]
Would be in line with other similar features.  As I commented on the GNB Bloodfest issue, though, features that replace a consumer with a 2-minute cooldown generator that's heavily tied to the burst window are, imo, both very poor features to use and borderline traps.  Those cooldowns are intended to be aligned with raid buffs and other cooldowns (such as Wildfire), not just used because you hit Hypercharge and didn't have enough heat for it at the time.  That's doubly the case now that Barrel Stabilizer also unlocks the extremely potent `Murica button (Full Metal Field)..

**Comment by fineanddandy26** [comment]
I think it's a little different in Machinist's case. The thing about Gunbreaker that makes it very different to most builder/spender jobs is that it's only building and spending up to 3 "points" - ergo, Bloodfest is not only mandatory but extremely crucial to maximizing damage during your burst.

Hypercharge for Machinist is as well, of course, but because Heat can go all the way to 100 and you get some on every GCD, not like Gunbreaker where you only get a bullet on the last hit, I think it's a lot less likely Hypercharge becomes a "trap" button.

Either way, even if it were to be a trap button, that's why I suggested it be an accessibility feature. Even with most of the Combos enabled Machinist is still one of the highest APM jobs.

**Comment by kaedys** [comment]
Well, there's two things about that.  One, Barrel Stabilizer doesn't even grant heat anymore, it grants a free Hypercharge (which is fully intended to be used with Wildfire during raid buffs), which is somewhat important, since the free Hypercharge buff has a duration and can be wasted without using it, unlike when it granted Heat directly.  Second, it grants access to Full Metal Field, which is _absolutely_ intended to be used with raid buffs.  So ya, Barrel Stabilizer is still very definitely mandatory for the burst phase, especially with that 900 potency guaranteed direct crit attached.

> Even with most of the Combos enabled Machinist is still one of the highest APM jobs.

Sure, but a combo that saves you one keypress every 120s, at the cost of very possibly using it at the wrong time and _wasting_ one or even _both_ of the benefits of it, isn't really a win in that book, imo.

**Comment by fineanddandy26** [comment]
Well it would be optional so if someone's not comfortable pressing it they wouldn't enable it, right?

**Comment by kaedys** [comment]
Sure, but that's true of any of our features.  Adding new ones still has to make _sense_.

Now, given that we have features similar to this already on other jobs, I think that bar is probably already reached, but that won't stop me from being critical of those features and seeing them as traps.  Or from _noting_ that they aren't recommended (and why) in their descriptions, for that matter... >.>

**Comment by fineanddandy26** [comment]
That's very fair. And admittedly despite being probably marked as an accessibility feature I don't think a beginner would benefit. However, I still think someone competent enough at Machinist to manage their meter well would get a lot of use out of the feature, just like someone good at Bullet management can use Bloodfest to great effect.

**Comment by kaedys** [comment]
Honestly, the worry I have with these features is less that they'll be used _early_ inadvertently, and more that players will enable them and then either not put that cooldown on their bar, or largely forget it exists, and then just use it whenever they happen to be low on resources and hit the button absentmindedly (or notice the cooldown has replaced the button).  And since nearly _all_ of those buttons now have powerful finishers attached to them, which absolutely should be and are _designed_ to be used during raid buffs, using them just whenever is both likely to waste that finisher (if the player hasn't enabled some other combo that automates using it), and even if it's used, likely to waste a significant portion of its potential due to using it outside the party buff window.

Like if a player goes into a dungeon boss fight already with 50 heat, and is used to using it when they press Hypercharge and are low on Heat at the time, ok, they throw out Wildfire and hopefully also use Hypercharge, but Barrel Stabilizer remains unused.  30s later, they go to Hypercharge again, or notice it's replaced Hypercharge, and use it, but completely forget that Full Metal Field even exists, much less is tied to that "yay free Hypercharge" button.

_That_ is what I mean by them being traps.  They're cooldowns, meant to be used strategically with raid buffs.  Granted, a decent number of casual players just don't use their cooldowns at all, and that's unfortunate.  This type of feature _could_, in theory, help them ever so slightly (but the value of a free Hypercharge thrown in outside raid buffs is actually very small).  But what would help far more is getting them to understand that literally every single job in the game has major damage cooldowns that align with a 60s or 120s interval, and that it's extremely important that those be used on the pull and as soon as they come off cooldown thereafter.  Just automating the usage behind the scenes isn't _really_ helping them.

Imagine if we had something similar for, say, Second Wind, just use it whenever it's available.  Probably would improve usage on players that never use it, but would waste _most_ of the benefit each time, and wouldn't _really_ be helping them at all, in the short _or_ long term.

**Comment by kaedys** [comment]
> However, I still think someone competent enough at Machinist to manage their meter well would get a lot of use out of the feature, just like someone good at Bullet management can use an easily pressable Bloodfest to great effect.

I would argue that anyone capable of optimizing the usage of it is 1) capable of hitting it on a separate button (ok, _probably_, physical accessibility really is a thing), and 2) is _much better served_ by having it on a separate button, both for tracking the cooldown and for being able to _hold_ it in situations where it makes sense (such as holding 2-minutes for a boss about to disappear, or you're at the end of a trash pack right before a boss in a dungeon, etc).

**Comment by kaedys** [comment]
> As for the Full Metal Field part... well, if a player isn't checking to make sure all their useable abilities are on their hotbar as is, in the nicest way possible, they're probably already a "lost cause". I don't think they care about being bare minimum optimal whatsoever, so their use case wouldn't really apply here.

I'd agree with you here _IF_ Barrel Stabilizer didn't automatically transform into Full Metal Field by default.  In fact, the design of _most_ of the DT additions is that they do _not_ need to have independent space on your hotbar, and instead simply replace the cooldown they triggers or enables them.  So not having Full Metal Field independently on your bar is both totally normal and to an extent _expected_.  Now, personally, I do have it separately on my bar, but for many of the other DT additions, I do not (ex. both of the new oGCDs for RDM, Wind and Fire Reply for MNK, all the new followups added to healers, etc).  So I don't think it's reasonable to _expect_ that users have these new actions on their bar, when the game pretty well expects that they won't me.  As a _key_ element of this, unlike other new actions, these ones are _not_ automatically added to your hotbar when you acquired them.

But really, Full Metal Field provides _way_ more damage contribution than a free Hypercharge, _especially_ a free one used randomly and not aligned with buffs and Wildfire.

> But that's why I specified the conditions in my original post.

I would actually argue that the conditions in your original post are more or less _built_ to waste the cooldown, since you advocated for only replacing it if it's off cooldown _and the user has less than 50 heat_.  Cooldowns like that are designed to be _immediately when available, excepting situations when you're intentionally holding it for something upcoming.  You did address that in your immediate follow-up, but I'd advocate that's really the only form that makes sense, always using it if it is off cooldown, not just if off CD and below 50 heat.  That's especially the case now that Barrel Stabilizer cannot cause you to overcap.

**Comment by MKhayle** [comment]
![357032336-d7973900-6f04-41ad-93ca-5396aff63699](https://github.com/user-attachments/assets/4b090f3e-17f9-4183-ae59-489f0002d6b7)
tl;dr?

**Comment by fineanddandy26** [comment]
TL;DR it should be an option because it's a standard option in other Job Combos but it's a trap option for unexperienced players like Empty Bloodfest Feature on Gunbreaker.

**Comment by MKhayle** [comment]
accessibility with a long description it is then

**Comment by kaedys** [comment]
Yep, that's a pretty sufficient TLDR.  I don't _like_ the feature, I think it's a trap, but it's also not any different from the slew of other features that work exactly like this, so I guess "meh".  Definitely needs a description note of why it's a bad idea, though.  This is the one I've added (locally, not yet PR'd) for GNB's Bloodfest:

```
NOTE: Bloodfest is intended to be used as a major damage cooldown aligned with party buffs and No Mercy.  It is not recommended to use it at random simply because you're out of powder gauge.
```

**Comment by MKhayle** [comment]
I haven't forgotten this request btw, sorry for the delay, it's coming (soon tm)

**Comment by MKhayle** [comment]
(soon tm) was a lie
